addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -1169,6 +1169,49 @@ QUnit.test('Post a message containing an email address followed by a mention on
     );
 });
 
+QUnit.test('Mention a partner with special character', async function (assert) {
+    assert.expect(1);
+
+    this.data['mail.channel'].records.push({ id: 11 });
+    this.data['res.partner'].records.push({
+        id: 1952,
+        email: "usatyi@example.com",
+        name: "Pynya's spokesman",
+    });
+    await this.start();
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 11,
+        model: 'mail.channel',
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+    document.querySelector('.o_ComposerTextInput_textarea').focus();
+    await afterNextRender(() => document.execCommand('insertText', false, "Dear "));
```

Comment:
This looks useless, so would drop this.

Reply:
For me, more actions in tests => more coverage, no?

Reply:
No hidden coverage in tests.

If you need to cover something, it has to be explicit so. For instance, if you need to test "Dear "+`@`-mention, you should make a test for precisely that... but don't add a test for that, because this looks useless right now. Adding more tests affect code maintenance, so the lesser the better if we can keep good working code. If we blindly aim for 100% coverage, we would have a huge amount of unmaintainable tests, which kills possibility to make necessary changes to our code fast.

This is a test for special character `'` in the name of partner that is being mentioned. It should be straight to the point. Of course not just `'` for name of partner, having something that looks like a name helps understanding the test at a quick glance.

Reply:
I deleted it, but I'm not sure that I agree with your argument about unmaintained tests. At least for this specific case. If we add typing "Dear " in every test, I'd still ok with that. I can't imagine an update that will drop possibility to start an email with "Dear "

Reply:
IMO tests are best when they map exactly to a spec. Spec is nice when it's super short and non-ambiguous. Doing so makes test short, super clear, and highly maintainable.

To come back to the "Dear ": it makes a user interaction that is totally useless. I wouldn't have minded it if the way to type something in composer was easy to read in test, but unfortunately that's not the case right now. This extra `afterNextRender(() => document.execCommand())` put more effort in understanding the test for something out-of-scope of test. That's why I much prefer to remove it.

Maybe in the future we will have improved expressiveness of user interactions in tests, so that having prefix "Dear " in message content doesn't make test far more noisy.